### PR TITLE
refactor: wallet, do not translate init options names

### PR DESCRIPTION
**Pull request by furszy**
Simple, and not interesting, refactor that someone has to do sooner or later. We are translating some init arguments names when those shouldn't be translated.

### CODE REVIEW

**Comment by hebasto** [comment]
Also #20404.

**Comment by furszy** [comment]
hmm k thanks @hebasto, I do agree with laanwj there. Some contextual information wouldn't be bad for this.
We could change a bit the error description by adding something like "invalid init argument value for %s bla bla" so translators can infer what will be placed in %s better.

Still, I'm good with any approach here. Pushed it merely because I found it while was working on other things in the wallet.

Also, I would keep this PR scoped to the wallet module only (same as someone could tackle other points of #20404 scoped to other area of the sources), mainly to not end up conflicting with many PRs.
Otherwise, the rebase work burden would be, by far, bigger than what it costs to do this tiny changes.

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -2909,7 +2909,7 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
     if (args.IsArgSet("-discardfee")) {
         std::optional<CAmount> discard_fee = ParseMoney(args.GetArg("-discardfee", ""));
         if (!discard_fee) {
-            error = strprintf(_("Invalid amount for -discardfee=<amount>: '%s'"), args.GetArg("-discardfee", ""));
+            error = strprintf(_("Invalid amount for %s=<amount>: '%s'"), "-discardfee", args.GetArg("-discardfee", ""));
```
Comment (luke-jr) [inline src/wallet/wallet.cpp:2912]: This message should probably be deduplicated to one location.

**Review by luke-jr — CHANGES_REQUESTED** [review]

**Comment by jarolrod** [comment]
concept ack, needs rebase

**Review by ryanofsky — APPROVED** [review]
Code review ACK 5fb8b862f455190dede1253dfd36cec7df14a7a8

There was some concern in https://github.com/bitcoin/bitcoin/pull/20404#issuecomment-729616853 that this change could lead to more questions from translators. But this PR is more limited than #20404, so this should be an easier place to start. Also it should be possible to provide context to translators through comments, and discussion in #20404 seems not to acknowledge benefits of the change for preventing translation errors.

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (furszy) [reply src/wallet/wallet.cpp:2912]: yeah, probably we should first move all the init args reads and validations to a separate file (`wallet_init.cpp` or something similar). Then cleanup all the string messages at once. So we don't continue mixing init stuff with wallet's internals.

**Comment by furszy** [comment]
great to see some movement here, rebased.

**Review by ryanofsky — APPROVED** [review]
Code review ACK e43a547a3674a31504a60ede9b4912e014a54139. Just rebased since last review.

**Review by hebasto — COMMENTED** [review]
Concept ACK on a general idea. Having some concerns about the approach, and wondering whether Transifex custom placeholders could be a better one.

See:
- https://docs.transifex.com/translation-checks/setting-translation-checks
- https://www.transifex.com/blog/2018/introducing-multiple-custom-placeholders-and-some-editor-goodies-for-translators/

**Comment by achow101** [comment]
ACK e43a547a3674a31504a60ede9b4912e014a54139

File: src/wallet/spend.cpp [hunk 2]
```diff
@@ -849,7 +849,7 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     }
     if (feeCalc.reason == FeeReason::FALLBACK && !wallet.m_allow_fallback_fee) {
         // eventually allow a fallback fee
-        return util::Error{_("Fee estimation failed. Fallbackfee is disabled. Wait a few blocks or enable -fallbackfee.")};
+        return util::Error{strprintf(_("Fee estimation failed. Fallbackfee is disabled. Wait a few blocks or enable %s."), "-fallbackfee")};
```
Comment (flack) [inline src/wallet/spend.cpp:852]: Strictly speaking, the middle sentence is a bit redundant (and also contains the option name)

Comment (hebasto) [reply src/wallet/spend.cpp:852]: For translators, it will be better to drop the middle sentence or put a place holder instead of the option name.

**Pull request merged**